src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -2284,6 +2319,21 @@ QueryPushdownTaskCreate(Query *originalQuery, int shardIndex,
 	List *selectPlacementList = NIL;
 	uint64 jobId = INVALID_JOB_ID;
 	uint64 anchorShardId = INVALID_SHARD_ID;
+	bool modifyWithSubselect = false;
+	RangeTblEntry *resultRangeTable = NULL;
+	Oid resultRelationOid = InvalidOid;
+
+	/*
+	 * If it is a modify query with sub-select, we need to have result relation
+	 * to obtain right lock on it.
```

Comment:
this comment is a bit misleading since we don't actually have any special logic to obtain a lock on the result relation, rather on all the other relations.